src/overlays/actors/ovl_Bg_Hidan_Hrock/z_bg_hidan_hrock.c

```diff
@@ -65,21 +69,174 @@ static ColliderTrisInit D_80889898 = {
+    Vec3f vertices[3];
+    f32 cosRotY;
+    f32 sinRotY;
+    s32 i;
+    s32 j;
+    CollisionHeader* collisionHeader;
+
+    collisionHeader = NULL;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->unk_16A = thisx->params & 0x3F;
+    thisx->params = (thisx->params >> 8) & 0xFF;
+    Collider_InitTris(globalCtx, &this->collider);
+    Collider_SetTris(globalCtx, &this->collider, thisx, &sTrisInit, this->colliderItems);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+
+    sinRotY = Math_SinS(thisx->shape.rot.y);
+    cosRotY = Math_CosS(thisx->shape.rot.y);
+
+    if (thisx->params == 0) {
+        sinRotY *= 1.5f;
+        cosRotY *= 1.5f;
+    }
+
+    for (i = 0; i < 2; i++) {
+        colliderElementInit = &sTrisInit.elements[i];
+
+        if (1) {
+            for (j = 0; j < 3; j++) {
+                Vec3f* vtx = &colliderElementInit->dim.vtx[j];
+
+                vertices[j].x = vtx->z * sinRotY + (thisx->home.pos.x + vtx->x * cosRotY);
+                vertices[j].y = vtx->y + thisx->home.pos.y;
+                vertices[j].z = vtx->z * cosRotY + (thisx->home.pos.z - vtx->x * sinRotY);
+            }
+        }
+        Collider_SetTrisVertices(&this->collider, i, &vertices[0], &vertices[1], &vertices[2]);
+    }
+
+    if (Flags_GetSwitch(globalCtx, this->unk_16A)) {
+        this->actionFunc = func_808894A4;
+        if (thisx->params == 0) {
+            thisx->world.pos.y -= 2800.0f;
+            thisx->uncullZoneForward = 3000.0f;
+        } else if (thisx->params == 1) {
+            thisx->world.pos.y -= 800.0f;
+        } else if (thisx->params == 2) {
+            thisx->world.pos.y -= 240.0f;
+        }
+    } else {
+        if (thisx->params == 0) {
+            thisx->flags |= 0x30;
+            thisx->uncullZoneForward = 3000.0f;
+        }
+        this->actionFunc = func_808896B8;
+    }
+
+    if (thisx->params == 0) {
+        CollisionHeader_GetVirtual(&D_0600D054, &collisionHeader);
+    } else {
+        CollisionHeader_GetVirtual(&D_0600D188, &collisionHeader);
+    }
+
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, collisionHeader);
+}
+
+void BgHidanHrock_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanHrock* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(globalCtx, &this->collider);
+}
+
+void func_808894A4(BgHidanHrock* this, GlobalContext* globalCtx) {
+}
+
+void func_808894B0(BgHidanHrock* this, GlobalContext* globalCtx) {
+    if (this->unk_168 != 0) {
+        this->unk_168--;
+    }
+
+    this->dyna.actor.world.pos.x =
+        (Math_SinS(this->dyna.actor.world.rot.y + (this->unk_168 << 0xE)) * 5.0f) + this->dyna.actor.home.pos.x;
+    this->dyna.actor.world.pos.z =
+        (Math_CosS(this->dyna.actor.world.rot.y + (this->unk_168 << 0xE)) * 5.0f) + this->dyna.actor.home.pos.z;
+
+    if (this->unk_168 % 4) {
+
+    } else {
```

Comment:
```suggestion
    if((this->unk_168 % 4) == 0) {
```
or failing that, `if(!(this->unk_168 % 4))`?